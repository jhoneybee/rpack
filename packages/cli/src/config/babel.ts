import * as Config from 'webpack-chain';

export const presetBabelReact = (config: Config) => {
    config
        .module
        .rule('babel-react')
        .test(/\.ts(x?)$/)
        .use('babel-react/loader')
        .loader('babel-loader')
        .options({
            presets: [
                ['@babel/preset-env', {
                    targets: 'defaults and not ie 11 and last 2 versions',
                }],
                '@babel/preset-react',
                '@babel/preset-typescript',
            ],
        })
        .end();
};
