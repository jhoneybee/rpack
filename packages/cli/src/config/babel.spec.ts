import * as Config from 'webpack-chain';
import { presetBabelReact } from './babel';

test('test presetBabelReact', () => {
    const config = new Config();
    presetBabelReact(config);
    expect(config.toConfig()).toMatchSnapshot();
});
