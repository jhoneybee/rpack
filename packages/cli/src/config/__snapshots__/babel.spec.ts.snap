// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test presetBabelReact 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.ts\\(x\\?\\)\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
            "options": Object {
              "presets": Array [
                Array [
                  "@babel/preset-env",
                  Object {
                    "targets": "defaults and not ie 11 and last 2 versions",
                  },
                ],
                "@babel/preset-react",
                "@babel/preset-typescript",
              ],
            },
          },
        ],
      },
    ],
  },
}
`;
